stl/src/syncstream.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// initialize syncstream mutex map

```

Comment:
We need to determine whether the import lib is the right place for the syncstream mutex map. Because the import lib is effectively statically linked into the user's binary, scenarios such as multiple user binaries (e.g. `meow.dll`, `purr.dll`, `cat.exe`) being linked to a single STL DLL (`msvcp140.dll`) would end up with separate syncstream mutex maps (one per user binary), which seems problematic.

Because these mutexes are global state, I suspect that we need a single copy of them per STL DLL, which means that this needs to be added to a satellite DLL. (For ease of development, the `atomic_wait` DLL recently added for C++20 is probably best to extend, despite the name - `/std:c++latest` is not yet ABI-frozen so we can add exports to that DLL, which will be massively easier than adding a new satellite DLL.)